addons/mail/static/src/messaging/component/activity_mark_done_popover/activity_mark_done_popover_tests.js

```diff
@@ -0,0 +1,285 @@
+    beforeEach: utilsBeforeEach,
+    start: utilsStart,
+} = require('mail.messaging.testUtils');
+
+QUnit.module('mail', {}, function () {
+QUnit.module('messaging', {}, function () {
+QUnit.module('component', {}, function () {
+QUnit.module('ActivityMarkDonePopover', {
+    beforeEach() {
+        utilsBeforeEach(this);
+        this.createActivityMarkDonePopoverComponent = async function (activity) {
+            const ActivityMarkDonePopoverComponent = components.ActivityMarkDonePopover;
+            ActivityMarkDonePopoverComponent.env = this.env;
+            this.component = new ActivityMarkDonePopoverComponent(null, {
+                activityLocalId: activity.localId,
+            });
+            await this.component.mount(this.widget.el);
+        };
+        this.start = async params => {
+            let { env, widget } = await utilsStart(Object.assign({}, params, {
+                data: this.data,
+            }));
+            this.env = env;
+            this.widget = widget;
+        };
+    },
+    afterEach() {
+        utilsAfterEach(this);
+        if (this.component) {
+            this.component.destroy();
+        }
+        if (this.widget) {
+            this.widget.destroy();
+        }
+        this.env = undefined;
+    },
+});
+
+QUnit.test('activity mark done popover simplest layout', async function (assert) {
+    assert.expect(6);
+
+    await this.start();
+    const activity = this.env.entities.Activity.create({
+        activity_category: 'not_upload_file',
+        can_write: true,
+        id:12,
+    });
+    await this.createActivityMarkDonePopoverComponent(activity);
+
+    assert.containsOnce(
+        document.body,
+        '.o_ActivityMarkDonePopover',
+        'Popover component should be present'
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_ActivityMarkDonePopover_feedback',
+        'Popover component should contain the feedback textarea'
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_ActivityMarkDonePopover_buttons',
+        'Popover component should contain the action buttons'
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_ActivityMarkDonePopover_doneScheduleNextButton',
+        'Popover component should contain the done & schedule next button'
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_ActivityMarkDonePopover_doneButton',
+        'Popover component should contain the done button'
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_ActivityMarkDonePopover_discardButton',
+        'Popover component should contain the discard button'
+    );
+});
+
+QUnit.test('activity with force next mark done popover simplest layout', async function (assert) {
+    assert.expect(6);
+
+    await this.start();
+    const activity = this.env.entities.Activity.create({
+        activity_category: 'not_upload_file',
+        can_write: true,
+        forceNext: true,
```

Comment:
Mix of case is strange, is one coming from the server and the other created in JS?

Reply:
I think it's a bug indeed.